src/main/java/com/commercetools/sync/categories/CategoryTypeDiff.java

```diff
@@ -0,0 +1,218 @@
+package com.commercetools.sync.categories;
+
+
+import com.commercetools.sync.services.TypeService;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryDraft;
+import io.sphere.sdk.categories.commands.updateactions.SetCustomField;
+import io.sphere.sdk.categories.commands.updateactions.SetCustomType;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.types.CustomFields;
+import io.sphere.sdk.types.CustomFieldsDraft;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class CategoryTypeDiff {
+
+    /**
+     * Compares the {@link CustomFields}, of an existing {@link Category}, to the {@link CustomFieldsDraft}, of a new
+     * {@link CategoryDraft}, and returns a {@link List<UpdateAction<Category>>} as a result. If no update action is
+     * needed, for example in the case where both the {@link CustomFields} and the {@link CustomFieldsDraft} are null,
+     * an empty {@link List<UpdateAction<Category>>} is returned. A {@link TypeService} instance is injected into the
+     * method to fetch the key of the existing category type from it's
+     * cache (see {@link CategoryTypeDiff#buildNonNullCustomFieldsActions(CustomFields, CustomFieldsDraft, TypeService)}).
+     * <p>
+     * <p>
+     * An update action will be added to the result list in the following cases:-
+     * <p>
+     * 1. If the new category's custom type is set, but existing category's custom type is not. A "setCustomType" update
```

Comment:
@lojzatran not sure about adding them in between the code of the method since now the list item numbers are generated by the html tags. However, in this specific method there are two comments in between the method that could help the reader.